staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/helpers.go

```diff
@@ -128,16 +128,18 @@ func HasCRDVersion(crd *CustomResourceDefinition, version string) bool {
 
 func GetCRDStorageVersion(crd *CustomResourceDefinition) string {
 	for _, v := range crd.Spec.Versions {
-		if v.Storage {
+		if IsCRDStorageVersion(&v) {
 			return v.Name
 		}
 	}
 	// This should not happened if crd is valid
 	return ""
 }
 
-func ConvertCRDVersion(crd *CustomResourceDefinition, newVersion string) *CustomResourceDefinition {
-	newCrd := crd.DeepCopy()
-	crd.TypeMeta.APIVersion = crd.GroupVersionKind().Group + "/" + newVersion
-	return newCrd
+func IsCRDVersionServed(version *CustomResourceDefinitionVersion) bool {
+	return version.Served == nil || *version.Served
```

Comment:
This nil case is odd. I would prefer to have `DisableServing` in the types to align the Golang default value and the semantical default.